sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sns/SnsIO.java

```diff
@@ -281,22 +281,33 @@ public Write<T> withRetryConfiguration(RetryConfiguration retry) {
     }
 
     /**
-     * Encode the full {@code PublishResult} object, including sdkResponseMetadata and
+     * Encode the full {@code PublishResponse} object, including sdkResponseMetadata and
      * sdkHttpMetadata with the HTTP response headers.
+     *
+     * @deprecated Writes fail fast in case of errors, no need to check headers.
```

Comment:
- Is this a change in behavior that we need to communicate to users?
- Will these functions be removed later?

Reply:
This is just a clarification why there's no need for a coder that includes HTTP response metadata, there's no change in behavior.
Plans is to remove these after a few releases.